ffi/src/lib.rs

```diff
@@ -328,6 +330,7 @@ impl ClientAdapter {
     ///
     /// For async clients, spawns the future and returns null immediately.
     /// For sync clients, blocks on the future and returns a `CommandResult`.
+    #[must_use]
```

Comment:
`#[must_use]` annotations force us not to ignore returned results. This prevents us from forgetting to handle a result and dropping errors on accident. Typically, Rust's `Result<T, E>` type already does this, but we create a custom `repr(C)` result type to return errors across the FFI boundary, which isn't `#[must_use]`. However, since we have to allocate the returned results, we return a raw pointer to that type, so marking it `#[must_use]` is meaningless. Thus, the only solution for now is to mark all functions that return `*mut CommandResult` as `#[must_use]`. (Type aliasing doesn't work here; try it yourself.)

Reply:
Can we set it globally for the entire project?

Reply:
No, just types and function returns. We don't want to do that for all functions and types anyway.